src/main/java/io/smartcat/ranger/core/parser/DataGenerator.java

```diff
@@ -0,0 +1,108 @@
+package io.smartcat.ranger.core.parser;
+
+import io.smartcat.ranger.core.CompositeValue;
+import io.smartcat.ranger.core.PrimitiveValue;
+import io.smartcat.ranger.core.Value;
+import io.smartcat.ranger.core.ValueProxy;
+import org.parboiled.Parboiled;
+import org.parboiled.parserunners.ReportingParseRunner;
+import org.parboiled.support.ParsingResult;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Generates data specified by configuration.
+ */
+public class DataGenerator {
+
+    private CompositeValue outputValue;
+
+    private DataGenerator(Map<String, ValueProxy<?>> proxyValues, String outputValueName) {
+        Value<?> val = proxyValues.get(outputValueName);
```

Comment:
`proxyValues` could be avoided by extracting the output value outside of the constructor.